rusk/src/lib/services/prover/stco.rs

```diff
@@ -0,0 +1,116 @@
+        let mut reader = &request.circuit_inputs[..];
+
+        if reader.len() != STCO_INPUT_LEN {
+            return Err(Status::invalid_argument(format!(
+                "Expected length {} got {}",
+                STCO_INPUT_LEN,
+                reader.len()
+            )));
+        }
+
+        let value = u64::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing value")
+        })?;
+        let r = JubJubScalar::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing 'r'")
+        })?;
+        let blinder = JubJubScalar::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing blinder")
+        })?;
+        let is_public = u64::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing is_public")
+        })? != 0;
+        let psk = PublicSpendKey::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing public spend key")
+        })?;
+        let pk_r = JubJubAffine::from_reader(&mut reader)
+            .map_err(|_| Status::invalid_argument("Failed deserializing pk_r"))?
+            .into();
+        let message = Message::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing message")
+        })?;
+        let crossover_blinder = JubJubScalar::from_reader(&mut reader)
+            .map_err(|_| {
+                Status::invalid_argument(
+                    "Failed deserializing crossover blinder",
+                )
+            })?;
+        let crossover = Crossover::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing crossover")
+        })?;
+        let fee = Fee::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing fee")
+        })?;
+        let contract_address =
+            BlsScalar::from_reader(&mut reader).map_err(|_| {
+                Status::invalid_argument(
+                    "Failed deserializing contract address",
+                )
+            })?;
+        let signature = Signature::from_reader(&mut reader).map_err(|_| {
+            Status::invalid_argument("Failed deserializing signature")
+        })?;
+
+        let derive_key = DeriveKey::new(is_public, &psk);
+
+        let stco_message = StcoMessage {
+            r,
+            blinder,
+            derive_key,
+            pk_r,
+            message,
+        };
+        let stco_crossover = StcoCrossover::new(crossover, crossover_blinder);
+
+        let mut circ = SendToContractObfuscatedCircuit::new(
+            value,
+            stco_message,
+            stco_crossover,
+            &fee,
+            contract_address,
+            signature,
+        );
+
+        let proof = circ
+            .prove(&*crate::PUB_PARAMS, &STCO_PROVER_KEY, b"dusk-network")
```

Comment:
if you don't use `lazy_static` but `Lazy` you should be able to use just the ref (without the deref), give a try.